backend/perf.js

```diff
@@ -80,275 +79,285 @@ function init(repo_loc, callback) {
-            var header = contents.header;
-            var times = contents.times;
-            if (times.length == 0) {
-                no_times += 1;
-                return;
+        let contents = JSON.parse(file_contents);
+        let header = contents.header;
+        let times = contents.times;
+        if (times.length == 0) {
+            no_times += 1;
+            continue;
+        }
+        let test_name = filename.substring(0, filename.indexOf("--"));
+
+        let date = new Date(header.date);
+        if (isNaN(date)) {
+            let date_components = [0, 0, 0, 0, 0, 0];
+            let actual = header.date.split("-").map(comp => parseInt(comp, 10));
+
+            for (let i = 0; i < actual.length; i++) {
+                date_components[i] = actual[i];
             }
-            var test_name = filename.substring(0, filename.indexOf('--'));
-
-            var date = new Date(header.date);
-            if (isNaN(date)) {
-                var date_components = [0, 0, 0, 0, 0, 0];
-                var actual_components = header.date.split('-');
-                for (i in actual_components) {
-                    date_components[i] = parseInt(actual_components[i], 10);
-                }
-                if (date_components[1] > 0) {
-                    date_components[1] -= 1;
-                }
 
-                date = new Date(date_components[0],
-                                date_components[1],
-                                date_components[2],
-                                date_components[3],
-                                date_components[4],
-                                date_components[5]);
+            if (date_components[1] > 0) {
+                date_components[1] -= 1;
             }
 
-            if (test_name == 'rustc') {
-                c_bootstraps += 1;
-                data['rustc'].insertOne(make_data(date, header, times, true));
-            } else {
-                if (benchmarks.indexOf(test_name) < 0) {
-                    benchmarks.push(test_name);
-                }
-                var key = data['benchmarks'].key({date: date});
-                if (key == null) {
-                    c_benchmarks += 1;
-                    data['benchmarks'].insertOne(make_data(date, header, times, false));
-                } else {
-                    c_benchmarks_add += 1;
-                    data['benchmarks'][key].by_crate[test_name] = make_times(times, false)[times[0].crate];
+            date = new Date(date_components[0],
+                            date_components[1],
+                            date_components[2],
+                            date_components[3],
+                            date_components[4],
+                            date_components[5]);
+        }
+
+        if (test_name == "rustc") {
+            c_bootstraps += 1;
+            data.rustc.insertOne(make_data(date, header, times, true));
+
+            for (let timing of times) {
```

Comment:
I'm a bit confused here - this seems to be an inlined version of `record_crate`, but that seemed to be called much later, so I'm not sure this is a safe move (it might well be, I just can't quite tell what is going on). Could you tell me what is happening please?